### PR TITLE
fix(sensor_download): add file operation support to module

### LINKED ISSUE #481: [falcon_install] Add ability to change permissions of the downloaded Falcon Sensor Installer
I use an Ansible playbook to create a golden image for AWS using the roles `falcon_install`, `falcon_configure` and `falcon_uninstall`. 

When I run the Ansible playbook, I get the following error message:

```
TASK [crowdstrike.falcon.falcon_install : CrowdStrike Falcon | Copy Sensor Installation Package to remote host (non-windows)] *******************************************************
fatal: [i-0da31599f96a90199_asg-metadefender-image-ubuntu22-mdtest-202404050828]: FAILED! =>
  msg: 'an error occurred while trying to read the file ''/var/tmp/falcon-sensor_7.13.0-16604_amd64.deb'': [Errno 13] Permission denied: b''/var/tmp/falcon-sensor_7.13.0-16604_amd64.deb''. [Errno 13] Permission denied: b''/var/tmp/falcon-sensor_7.13.0-16604_amd64.deb'''
```

The reason for this error message is a global `umask 0077` on the Ansible control node. 
After downloading the Falcon Sensor Installer via API, the file has the following permissions:

```
-rw------- 1 root root 58047924 Apr  5 08:40 /var/tmp/falcon-sensor_7.13.0-16604_amd64.deb
```

Although the Ansible playbook is executed with `become: true`, the Ansible task cannot access the file to copy it to the target host. 

I could probably work around the problem by replacing the roles with the appropriate modules, but it would be more convenient to run an optional task between `Download Falcon Sensor Installation Package (local)` and `Copy Sensor Installation Package to remote host (non-windows)` that changes the permissions of the file. 

https://github.com/CrowdStrike/ansible_collection_falcon/blob/main/roles/falcon_install/tasks/api.yml#L63-L71
https://github.com/CrowdStrike/ansible_collection_falcon/blob/main/roles/falcon_install/tasks/api.yml#L63-L71

Something like:

```
- name: CrowdStrike Falcon | Permissions of Falcon Sensor Installation Package (local)
  ansible.builtin.file:
    path: "{{ falcon_sensor_download.path }}"
    mode: "{{ falcon_api_sensor_download_mode }}"
  when: falcon_api_sensor_download_mode != ""
```

### PR DESCRIPTION
**Pull request by carlosmmatos**
Fixes #481

This PR introduces the built-in files AnsibleModule helper features to manage file operations such as mode, owner, group, etc.

### CODE REVIEW

File: plugins/modules/sensor_download.py [hunk 1]
```diff
@@ -65,6 +66,13 @@
     hash: "1234567890123456789012345678901234567890123456789012345678901234"
     dest: "/tmp/windows"
     name: falcon-sensor.exe
+
+- name: Download the Falcon Sensor Installer to a temporary directory and set permissions
+  crowdstrike.falcon.sensor_download:
+    hash: "1234567890123456789012345678901234567890123456789012345678901234"
+    mode: "0755"
```
Comment (dhoppe) [inline plugins/modules/sensor_download.py:73]: @carlosmmatos One last question. Are you sure about these permissions? Using 0644 should be sufficient.

**Review by redhatrises — CHANGES_REQUESTED** [review]
Please update the file mode example, 755 looks weird for a non-executable file.

File: plugins/modules/sensor_download.py (hunk 1 above)
Comment (carlosmmatos) [reply plugins/modules/sensor_download.py:73]: It was an example and clearly I was not thinking right last night.. I will update that to 0644

**Review by redhatrises — APPROVED** [review]

**Comment by dhoppe** [comment]
@carlosmmatos I am glad that the module `sensor_download` is now able to change the `owner`, `group` and `mode`, but the role does not yet have the corresponding variables / defaults.

**Comment by carlosmmatos** [comment]
Dope 🤦🏻‍♂️. You're right, I was caught up in the process of updating the sensor module that I forgot that needs to be applied to the role. Apologies for that, I will have that out by end of day.

**Pull request merged**